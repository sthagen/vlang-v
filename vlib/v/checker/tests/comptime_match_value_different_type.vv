fn match_value_different_type() {
	x := 100
	$match x {
		100 {
			println('int 100')
		}
		'100' {
			println('string 100')
		}
	}
}

fn match_type_different_literal() {
	$match $int {
		1 {
			println('IntegerLiteral')
		}
		'1' {
			println('StringLiteral')
		}
		true {
			println('BoolLiteral')
		}
	}
}
