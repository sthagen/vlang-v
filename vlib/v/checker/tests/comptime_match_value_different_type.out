vlib/v/checker/tests/comptime_match_value_different_type.vv:7:3: error: can not matching a string value(`'100'`) in a non string type `$match`, `x` type is `int`
    5 |             println('int 100')
    6 |         }
    7 |         '100' {
      |         ~~~~~
    8 |             println('string 100')
    9 |         }
vlib/v/checker/tests/comptime_match_value_different_type.vv:18:3: error: can not matching a string value(`'1'`) in a non string type `$match`, `$int` type is `int`
   16 |             println('IntegerLiteral')
   17 |         }
   18 |         '1' {
      |         ~~~
   19 |             println('StringLiteral')
   20 |         }
